### PR TITLE
[ARM] Specifically delineate between different GPRCS2 positions

### PR DESCRIPTION
**Pull request by pestctrl**
Currently, the relative position of GPRCS2 (with respect to other instructions in the prologue of a function) can be different depending on the type of ARMSubtarget::PushPopSplitVariant.

When the PushPopSpiltVariant is SplitR11WindowsSEH, GPRCS2 comes `after` both GPRCS1 and DPRCS2:

GPRCS1
DPRCS1
GPRCS2

However, in all other cases, GPRCS2 comes `before` DPRCS1, like so:

GPRCS1
GPRCS2
DPRCS1

This makes the MI walking code in ARMFrameLowering::emitPrologue a bit confusing. If GPRCS2Size is non-zero, we also have to check the PushPopSplitVariant to know if we will encounter the DPRCS1 push instruction first or the GPRCS2 push instruction first.

This commit changes to SplitR11WindowsSEH such that the spill area is as follows:

GPRCS1
DPRCS1
GPRCS3

This disambiguates a lot of the ARMFrameLowering.cpp MI traversal code.

### CODE REVIEW

File: llvm/lib/Target/ARM/ARMFrameLowering.cpp
Comment (pestctrl) [inline llvm/lib/Target/ARM/ARMFrameLowering.cpp:1095]: I left these asserts here to serve as reminders of what `GPRCS2Size > 0` and `GPRCS3Size > 0` likewise imply, but someone doesn't like them being here, I can remove them. Checking for GPRCS2Size/GPRCS3Size provides all the frame code needs to know.

File: llvm/lib/Target/ARM/ARMFrameLowering.cpp
Comment (ostannard) [inline llvm/lib/Target/ARM/ARMFrameLowering.cpp:2158]: Can this `if` be removed now that the blocks are always pushed in the same order?

File: llvm/lib/Target/ARM/ARMFrameLowering.cpp
Comment (ostannard) [inline llvm/lib/Target/ARM/ARMFrameLowering.cpp:2220]: Can this `if` be removed now that the blocks are always pushed in the same order?

File: llvm/lib/Target/ARM/ARMFrameLowering.cpp
Comment (ostannard) [inline llvm/lib/Target/ARM/ARMFrameLowering.cpp:1226]: I think it would be better to include `GPRCS2Size` here for consistency, or at least assert that it is zero, so that this will be correct of we ever start using GPRCS2 and GPRCS3 together.

File: llvm/lib/Target/ARM/ARMFrameLowering.cpp
Comment (pestctrl) [reply llvm/lib/Target/ARM/ARMFrameLowering.cpp:2220]: Right! Good idea.

File: llvm/lib/Target/ARM/ARMFrameLowering.cpp
Comment (pestctrl) [reply llvm/lib/Target/ARM/ARMFrameLowering.cpp:1226]: I agree with including `GPRCS2Size` here.

**Review by ostannard — APPROVED** [review]
Thanks, LGTM, assuming you apply the code_formatter bot's suggested changes before merging.

**Pull request merged**
